File: Exavision.Seasense/Client/src/app/api/ws/ws-turret-move-absolute-request.ts
import { PanTiltPosition } from "./pan-tilt-position";
import { SpeedVector } from "./speed-vector";
import { WsRequest } from "./ws-request";

export interface WsTurretMoveAbsoluteRequest extends WsRequest {  
  materialId: string;
  unitId: string;
  position: PanTiltPosition;
}

File: Exavision.Seasense.Api.WebSocket/Turret/WsTurretMoveAbsoluteRequest.cs
using Exavision.Seasense.Api.WebSocket.Core;
using Exavision.Seasense.Shared.Types;

namespace Exavision.Seasense.Api.WebSocket.Turret {
    public class WsTurretMoveAbsoluteRequest : WsRequest {
        public string UnitId { get; set; }
        public string MaterialId { get; set; }
        public PanTiltPosition Position { get; set; }
    }
}
